### PR TITLE
Fix HiRadix host cache load_back crash and internal-node eviction

### PR DESCRIPTION
**Pull request by chivalryq**
## Motivation

HiRadix host-cache load-back can crash when an evicted node is still treated as host-recoverable after its `host_value` has been freed by host eviction. In that state `load_back()` may call `len()` or `torch.cat()` on `None`, terminating the scheduler.

The same host eviction path also detached internal radix nodes when freeing host pages, which could orphan their child subtree even though the internal node was still needed as a prefix anchor.

## Modifications

- Protect every evicted node's host pages during `load_back()` and snapshot host segment lengths before restoring device pages.
- Add `TreeNode.storage_backed` to distinguish internal host nodes that can safely shed host pages because they are recoverable from L3 storage.
- Preserve radix subtrees when evicting host pages from storage-backed internal nodes; only true leaves are removed from the tree.
- Maintain storage-backed state on backup completion, L3 prefetch insertion, node split, storage clear, and backend detach.
- Stop host-prefix matching at the first host gap so descendant host pages are not counted as a contiguous loadable prefix.

Internal node host-page shedding does not emit `BlockRemoved(CPU)` because the block is not deleted from the radix tree; the subtree stays reachable and a host gap will trigger L3 recovery instead of router removal.

## Accuracy

The new tests cover host-evictable predicates, internal storage-backed host eviction, storage clear invalidation, host-gap matching, L3 prefetch insertion into storage-only nodes, duplicate host-page accounting after a host gap, and host protection during `load_back()`.

## Tests

- `python3 -m py_compile python/sglang/srt/mem_cache/hiradix_cache.py python/sglang/srt/mem_cache/radix_cache.py test/registered/unit/mem_cache/test_radix_cache_unit.py`
- `git diff --check upstream/main..HEAD`

Target unittest is expected to run in CI: `test/registered/unit/mem_cache/test_radix_cache_unit.py::TestHiRadixCacheHostState`.

---
### CI States

Latest PR Test (Base): :x: [Run #27137270652](https://github.com/sgl-project/sglang/actions/runs/27137270652)
Latest PR Test (Extra): :x: [Run #27137270371](https://github.com/sgl-project/sglang/actions/runs/27137270371)

### CODE REVIEW

**Comment by chivalryq** [comment]
Temporarily closing and reopening to retrigger PR CI after marking ready for review; the first run was blocked while the PR was draft.

**Comment by hzh0425** [comment]
Please provide a Minimal reproducible case—this might not be the root cause.

**Comment by chivalryq** [comment]
I'll post a repro. This is a fuse of several issue, I'll try to give repros for each of them and maybe split into several PRs.
Update: leaving a comments on nixl roadmap issue: https://github.com/sgl-project/sglang/issues/26693#issuecomment-4650907824
